### PR TITLE
Update sampler docs for unknown kwarg handling

### PR DESCRIPTION
**Pull request by hhtong**

### CODE REVIEW

File: dimod/core/sampler.py [hunk 1]
```diff
@@ -23,6 +23,10 @@
 or :meth:`~.Sampler.sample_qubo`. The :class:`.Sampler` class provides the complementary
 methods as mixins and ensures consistent responses.
 
+All sample methods must be able to accept unknown kwargs with a raised warning.
```
Comment (JoelPasvolsky) [inline dimod/core/sampler.py:26]: What about a slight rephrasing to something like, 
"Implemented sample methods must accept, and warn on, unknown keyword arguments ``**kwargs``.

File: dimod/core/sampler.py [hunk 2]
```diff
@@ -23,6 +23,10 @@
...
+This means that all sample methods must have the `**kwargs` parameter and
```
Comment (JoelPasvolsky) [inline dimod/core/sampler.py:27]: "... parameter and call the :meth:`~.Sampler.remove_unknown_kwargs` method"?

File: dimod/core/sampler.py [hunk 3]
```diff
@@ -115,6 +121,8 @@ class Sampler(metaclass=SamplerABCMeta):
     Provides all methods :meth:`~.Sampler.sample`, :meth:`~.Sampler.sample_ising`,
     :meth:`~.Sampler.sample_qubo` assuming at least one is implemented.
 
+    Also includes utility method :meth:`~.Sampler.remove_unknown_kwargs`, which
+    should be called in each sample method.
```
Comment (JoelPasvolsky) [inline dimod/core/sampler.py:125]: "... each implemented sample method"?

File: dimod/core/sampler.py (hunk 1 above)
Comment (hhtong) [reply dimod/core/sampler.py]: That sounds better, thanks!

File: dimod/core/sampler.py [hunk 4]
```diff
@@ -23,6 +23,10 @@
 or :meth:`~.Sampler.sample_qubo`. The :class:`.Sampler` class provides the complementary
 methods as mixins and ensures consistent responses.
 
+Implemented sample methods must accept, and warn on, unknown keyword arguments
+`**kwargs`. This means that all sample methods must have the `**kwargs`
+parameter and call the :meth:`~.Sampler.remove_unknown_kwargs` method.
```
Comment (arcondello) [inline dimod/core/sampler.py:28]: The user technically does not have to call the method. Maybe 
```
...must have `**kwargs` parameter. The :meth:`~.Sampler.remove_unknown_kwargs` convenience method is provided for this purpose.
```
or something like that?

Comment (hhtong) [reply dimod/core/sampler.py]: That's true. I've updated with this change.

**Review by arcondello — APPROVED** [review]

File: dimod/core/sampler.py [hunk 5]
```diff
@@ -115,6 +122,8 @@ class Sampler(metaclass=SamplerABCMeta):
     Provides all methods :meth:`~.Sampler.sample`, :meth:`~.Sampler.sample_ising`,
     :meth:`~.Sampler.sample_qubo` assuming at least one is implemented.
```
Comment (JoelPasvolsky) [inline dimod/core/sampler.py:124]: According to comment https://github.com/dwavesystems/dimod/pull/799#discussion_r577249029, the implementation does not have to call the method but below you say ", which  should be called in each implemented sample method"

**Review by JoelPasvolsky — APPROVED** [review]

**Pull request merged**